### PR TITLE
Hitl dashboard server

### PR DESCRIPTION
**Pull request by mialsy**
# Description

Added a prototype for Hitl dashboard app, included:
- Backend: a backend based on flask and socket. Run on localhost:5000 by default.
- Frontend: a react based frontend - currently only have button for user interaction. Default run on localhost:3000.

Please note this dashboard app is currently only in development build and is not optimized.

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

Now we can see three simple APIs provided as shown in the gif below:
backend_api_demo

# Testing

- Unit test for backend helper methods to interact with aws  (see `dashboard_aws_helper_test.py` file)
- Manually tested frontend - backend interaction, as shown in the demo above.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [x] I have made changes to existing documentation where needed.
- [x] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

File: droidlet/dashboard/web/config/webpack.config.js [hunk 1]
```diff
@@ -53,6 +53,10 @@ const sassModuleRegex = /\.module\.(scss|sass)$/;
 // This is the production and development configuration.
 // It is focused on developer experience, fast rebuilds, and a minimal bundle.
 module.exports = function(webpackEnv) {
+  watchOptions: {
```
Comment (snyxan) [inline droidlet/dashboard/web/config/webpack.config.js:56]: HiTL dashboard is different from Droidlet dashboard. This PR shouldn't touch any files under `droidlet/dashboard`, right?

File: droidlet/dashboard/web/src/StateManager.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ class StateManager {
   refs = [];
   socket = null;
   worldSocket = null;
-  default_url = "http://localhost:8000";
+  default_url = "http://localhost:5000";
```
Comment (snyxan) [inline droidlet/dashboard/web/src/StateManager.js:54]: same as above

File: droidlet/dashboard/web/config/webpack.config.js (hunk 1 above)
Comment (mialsy) [reply droidlet/dashboard/web/config/webpack.config.js:56]: Yeah, thanks for pointing this out, I was testing on the previous dashboard before, but forgot to change it back. Let me revert this change.

**Comment by snyxan** [comment]
Overall LGTM. The newly added unittest is failing due to the lack of credentials. Please set those up and merge.

**Comment by snyxan** [comment]
Tried to run it locally and backend doesn't seem to receive requests from frontend (i.e. No log msg print out on server side when click buttons in dashboard, nor any data shown after clicking). Is there any specific setup I need to do in additional to the steps in README?

**Comment by mialsy** [comment]
> Tried to run it locally and backend doesn't seem to receive requests from frontend (i.e. No log msg print out on server side when click buttons in dashboard, nor any data shown after clicking). Is there any specific setup I need to do in additional to the steps in README?

A few problem that I have in mind:
1. I wonder if you are able to start the server, i.e. do you see the server running? There is a dependency on `flask-socketio`  - I just realized that I only installed it on my own conda environment and I will need to update the dependency file.
2. You probably have it set up, but just to confirm that you have ssh tunneling for both frontend and backend? I did not add this to readme as I am assuming you can also run this on your local machine with the current conda environment. 
3. Can I see you console log from the browser? Or click on the `console log` button to check the socket connection (it will show info including if the connection is true on the console log). I encountered it a few times when restoring ssh connection to devfair, and there are CORS issues when using the same browser window, and it will go away if you quit the browser and reopen it can be fixed.

**Comment by snyxan** [comment]
> > Tried to run it locally and backend doesn't seem to receive requests from frontend (i.e. No log msg print out on server side when click buttons in dashboard, nor any data shown after clicking). Is there any specific setup I need to do in additional to the steps in README?
> 
> Hi @snyxan A few possible causes that I have in mind:
> 
> 1. I wonder if you are able to start the server, i.e. do you see the server running? There is a dependency on `flask-socketio`  - I just realized that I only installed it on my own conda environment and I will need to update the dependency file.

I can start the server, but there's nothing printed out in console.

> 2. You probably have it set up, but just to confirm that you have ssh tunneling for both frontend and backend? I did not add this to readme as I am assuming you can also run this on your local machine with the current conda environment.

I'm testing it on my local machine, i.e. frontend, backend & browser are in the same host so no need to do ssh tunneling.

> 3. Can I see your console log from the browser? Or click on the `console log` button to check the socket connection (it will show info including if the connection is true on the console log). I encountered it a few times when restoring ssh connection to devfair, and there are CORS issues when using the same browser window, and it will go away if you quit the browser and reopen it can be fixed.

When I click on `console log` the following content showed up: 

```
Socket {connected: true, receiveBuffer: Array(0), sendBuffer: Array(0), ids: 0, acks: {…}, …}
```
I assume that indicated successful socket connection.

> 
> If none of this works, can I ask you to send me a screenshot of your console, or I can schedule a meeting to see what is happening on your machine?

**Comment by mialsy** [comment]
> > > Tried to run it locally and backend doesn't seem to receive requests from frontend (i.e. No log msg print out on server side when click buttons in dashboard, nor any data shown after clicking). Is there any specific setup I need to do in additional to the steps in README?
> > 
> > 
> > Hi @snyxan A few possible causes that I have in mind:
> > 
> > 1. I wonder if you are able to start the server, i.e. do you see the server running? There is a dependency on `flask-socketio`  - I just realized that I only installed it on my own conda environment and I will need to update the dependency file.
> 
> **I can start the server, but there's nothing printed out in console.**
> 
> > 2. You probably have it set up, but just to confirm that you have ssh tunneling for both frontend and backend? I did not add this to readme as I am assuming you can also run this on your local machine with the current conda environment.
> 
> **I'm testing it on my local machine, i.e. frontend, backend & browser are in the same host so no need to do ssh tunneling.**
> 
> > 3. Can I see your console log from the browser? Or click on the `console log` button to check the socket connection (it will show info including if the connection is true on the console log). I encountered it a few times when restoring ssh connection to devfair, and there are CORS issues when using the same browser window, and it will go away if you quit the browser and reopen it can be fixed.
> 
> **When I click on `console log` the following content showed up:**
> 
> ```
> Socket {connected: true, receiveBuffer: Array(0), sendBuffer: Array(0), ids: 0, acks: {…}, …}
> ```
> 
> **I assume that indicated successful socket connection.**
> 
> > If none of this works, can I ask you to send me a screenshot of your console, or I can schedule a meeting to see what is happening on your machine?

Note: the issue was coming from using different version of `flask` and `flask-socketio`; updated README.MD regarding the version required to run the server.

**Review by snyxan — APPROVED** [review]
AWS CRED issue has been resolved. Please rebase on main and merge it in once unittests passed.

**Pull request merged**